cpp/src/arrow/dataset/file_parquet.h

```diff
@@ -76,35 +78,20 @@ class ARROW_DS_EXPORT ParquetFileFormat : public FileFormat {
     /// \defgroup parquet-file-format-reader-properties properties which correspond to
     /// members of parquet::ReaderProperties.
     ///
-    /// We don't embed parquet::ReaderProperties directly because we get memory_pool from
-    /// ScanOptions at scan time and provide differing defaults.
-    ///
     /// @{
-    bool use_buffered_stream = false;
-    int64_t buffer_size = 1 << 13;
     std::shared_ptr<parquet::FileDecryptionProperties> file_decryption_properties;
     /// @}
 
     /// \defgroup parquet-file-format-arrow-reader-properties properties which correspond
     /// to members of parquet::ArrowReaderProperties.
     ///
-    /// We don't embed parquet::ReaderProperties directly because we get batch_size from
-    /// ScanOptions at scan time, and we will never pass use_threads == true (since we
-    /// defer parallelization of the scan). Additionally column names (rather than
-    /// indices) are used to indicate dictionary columns.
+    /// We don't embed parquet::ReaderProperties directly because column names (rather
+    /// than indices) are used to indicate dictionary columns, and other options are
+    /// deferred to scan time.
     ///
     /// @{
     std::unordered_set<std::string> dict_columns;
-    bool pre_buffer = false;
-    arrow::io::CacheOptions cache_options = arrow::io::CacheOptions::Defaults();
-    arrow::io::IOContext io_context;
     /// @}
-
-    /// EXPERIMENTAL: Parallelize conversion across columns. This option is ignored if a
-    /// scan is already parallelized across input files to avoid thread contention. This
-    /// option will be removed after support is added for simultaneous parallelization
-    /// across files and columns.
-    bool enable_parallel_column_conversion = false;
   } reader_options;
```

Comment:
I think now that `FileFormat::default_fragment_scan_options` is available and both IPC and CSV file formats embed options structs whole it'd be best to just follow that pattern with parquet: let ParquetFragmentScanOptions hold a ReaderProperties whole.

`dict_columns` will need to remain here for now since it affects the schema of the scanned dataset, but once we can (for example) project a string column to dictionary<int32, string> I'd expect this option to be deprecated in favor of automatically pushing the projection-to-dictionary down to the parquet reader with `ArrowReaderOptions::set_read_dictionary`.

`file_decryption_properties` may also need to be replicated, though I'm not sure how/if anyone is using encryption + datasets